File: convertedToIOS5/p741p758XML/p741p758XML/Person.m

#import "Person.h"


@implementation Person
@synthesize lastName, firstName;


- (void)encodeWithCoder:(NSCoder *)encoder {
    //[super encodeWithCoder: encoder];
    [encoder encodeObject:self->lastName forKey:@"last"];
    [encoder encodeObject:self->firstName forKey:@"first"];
}

- (id) initWithCoder:(NSCoder *)decoder {
    //self = [super initWithCoder: decoder];
    self = [super init];
    self->lastName = [decoder decodeObjectForKey:@"last"];
    self->firstName = [decoder decodeObjectForKey:@"first"];
    return self;
}

- (NSString*) description {
    return [NSString stringWithFormat: @"%@ %@", firstName, lastName];
}

@end

File: iOS10bookexamples/bk2ch23p798basicFileOperations/ch36p1053basicFileOperations/Person.swift


import UIKit

class Person: NSObject, NSCoding {
    var firstName : String
    var lastName : String
    
    override var description : String {
        return self.firstName + " " + self.lastName
    }
    
    init(firstName:String, lastName:String) {
        self.firstName = firstName
        self.lastName = lastName
        super.init()
    }
    
    func encode(with coder: NSCoder) {
        // do not call super in this case
        coder.encode(self.lastName, forKey: "last")
        coder.encode(self.firstName, forKey: "first")
    }
    
    required init(coder: NSCoder) {
        self.lastName = coder.decodeObject(forKey:"last") as! String
        self.firstName = coder.decodeObject(forKey:"first") as! String
        // do not call super init(coder:) in this case
        super.init()
    }

}
